src/nsqdconnection.coffee

```diff
@@ -246,19 +251,23 @@ class ConnectionState extends NodeState
   constructor: (@conn) ->
     super
       autostart: true,
-      initial_state: 'CONNECTING'
+      initial_state: 'INIT'
       sync_goto: true
 
     @identifyResponse = null
 
   log: (message) ->
-    @conn.debug "#{@current_state_name}"
+    @conn.debug "#{@current_state_name}" unless @current_state_name is 'INIT'
```

Comment:
It addresses the false debug logging that a new connection was being created. By introducing a new state machine step, I can filter out those messages.